gems/smithy-schema/lib/smithy-schema/document_utils.rb

```diff
@@ -0,0 +1,231 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'time'
+
+module Smithy
+  module Schema
+    # @api private
+    # Document Utilities to help (de)construct data to/from Smithy document
+    module DocumentUtils
```

Comment:
I could separate them into different modules/classes like Builder, Parsers (maybe call it something else) down the road if this file is bloated.